src/pages/vrf/v2/examples/lootbox.mdx

```diff
@@ -0,0 +1,309 @@
+
+Next, switch accounts and open the lootbox as a user to test the contract.
+
+## Open the lootbox
+
+Once the contract is deployed and the start timestamp is reached, users can start opening the lootbox. The amount of rewards they receive depends on the amount of units they open by specifying the `amountToOpen` parameter and paying the corresponding fee.
+
+You can control access to the lootbox with two modes: private and public. Use these modes to determine which users can open the lootbox. The owner account can update this setting after deployment, so you can test the opening functionality privately before making the lootbox available to the public.
+
+### Private mode
+
+In this mode, the lootbox is only open to addresses on the allow list by calling the `privateOpen` function and providing Merkle proof for the address. To generate it, see [merkletreejs](https://github.com/merkletreejs/merkletreejs).
+
+The allow list is set on contract deployment and can be changed by calling the `setWhitelistRoot` function from the owner account. The private mode can be enabled or disabled at any time by calling the `setPrivateOpen` function from the owner account.
+
+### Public mode
+
+When the private mode is disabled, anyone can open the lootbox by calling the `publicOpen` function. It will do the same thing as the `privateOpen` function but without the Merkle proof.
+
+## Randomness
+
+The contract uses [Chainlink VRF](https://vrf.chain.link) to generate randomness which is used to determine the rewards the user receives.
+
+Because the randomness is generated off-chain, the contract will not be able to transfer the rewards immediately. Instead, it will store the request and once the randomness is received, the user or anyone else can call the `claimRewards` function to transfer the rewards to the user.
+
+The lootbox creator can also call the `claimRewards` function and improve the user experience by transferring the rewards to the user immediately. You can automate this further by using [Chainlink Automation](https://automation.chain.link/).
+
+## Claim rewards
+
+The rewards for an open request can be claimed by calling the `claimRewards` function and passing the opener address as the parameter. This will transfer the rewards to the opener address.
+
+The claim function can only be called after the randomness is fulfilled. To check this, call the `canClaimRewards` function and pass the opener address as the parameter.
+
+One address can only have one open request at a time. If you try to open the lootbox again before the previous request is fulfilled, the transaction reverts with a `PendingOpenRequest` error.
+
+## Withdraw funds
+
+At any time, the owner can withdraw funds from the collected fees by calling the `withdraw` function. By doing so, the contract balance will be transferred to the owner account.
```

Comment:
Add a code analysis section. Similar to the template: https://github.com/smartcontractkit/documentation/blob/7d8b449de01739814f72532d0163e9fbfdf1cc59/templates/tutorial.mdx#L128

Reply:
Added.